src/core.js

```diff
@@ -217,7 +217,11 @@ jQuery.extend( {
 		// that can be coerced to finite numbers (gh-2662)
 		var type = jQuery.type( obj );
 		return ( type === "number" || type === "string" ) &&
-			( obj - parseFloat( obj ) + 1 ) >= 0;
+
+			// parseFloat NaNs numeric-cast false positives ("")
+			// ...but misinterprets leading-number strings, particularly hex literals ("0x...")
+			// subtraction forces infinities to NaN
+			!isNaN( obj - parseFloat( obj ) );
```

Comment:
@gibson042 digging through the tickets, you proposed the change to be `!isNaN( obj - parseFloat( obj ) )` a few years back. But you finally went with `( obj - parseFloat( obj ) + 1 ) >= 0;`. Can you remember any reason?